Astronomer Enterprise v0.23 Release Notes (#203)
astronomer/docs: cloud/stable/06_resources/01_release-notes.md

## Astronomer v0.23

### v0.23.9

Release Date: January 18, 2021

> This release was exclusively infrastructure maintenance in preparation for Astronomer Enterprise v0.23.9 release. Stay tuned for v0.24!

Reviewer: I dont think we should say stay tuned for 0.24 .. Dont know when we will ship that just yet.

> This release was exclusively infrastructure maintenance in preparation for Astronomer Enterprise v0.23.9 release. Stay tuned for v0.24!

Reviewer 2: ```suggestion
This release was exclusively infrastructure maintenance for the release of Astronomer Enterprise v0.23. Stay tuned for v0.24!
```

- Enforce that all users migrate to Astronomer Certified 1.10.14 before upgrading to 2.0
- Support Airflow's ["upgrade check"](https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html) in the Astronomer CLI (`$ astro dev upgrade-check`)

> To get started, read [Upgrade to Apache Airflow 2.0 on Astronomer](https://www.astronomer.io/docs/cloud/stable/customize-airflow/upgrade-to-airflow-2). The migration path to Airflow 2.0 on Astronomer follows all recommendations published by the Apache Airflow project.

Reviewer 2: Not seeing a link to these "published" recommendations here or in the Upgrade doc, maybe we should cut that word lest it send someone on a wild goose chase.

Author: Ah yea hm. Those recommendations are here: https://airflow.apache.org/docs/apache-airflow/stable/upgrading-to-2.html

I'll link to them!

> To get started, read [Upgrade to Apache Airflow 2.0 on Astronomer](https://www.astronomer.io/docs/cloud/stable/customize-airflow/upgrade-to-airflow-2). The migration path to Airflow 2.0 on Astronomer follows all recommendations published by the Apache Airflow project.

Author: ```suggestion
To get started, read [Upgrade to Apache Airflow 2.0 on Astronomer](https://www.astronomer.io/docs/cloud/stable/customize-airflow/upgrade-to-airflow-2). The migration path to Airflow 2.0 on Astronomer follows [recommendations published by the Apache Airflow project](https://airflow.apache.org/docs/apache-airflow/stable/upgrading-to-2.html).
```

For more information, refer to ["Manage Airflow Versions"](https://www.astronomer.io/docs/cloud/stable/customize-airflow/manage-airflow-versions/) or [this forum post](https://forum.astronomer.io/t/docker-hub-rate-limit-error-toomanyrequests-you-have-reached-your-pull-rate-limit/887).

#### Improved Celery Worker Update Strategy

> Astronomer v0.23 includes an improvement to the process by which new Celery Workers are created and begin to pick up tasks following a deploy to an Airflow Deployment on Astronomer.

Reviewer 2: ```suggestion
Astronomer v0.23 includes an improvement to how Celery Workers are created when deploying to an Airflow Deployment on Astronomer.
```
We don't describe how the actual task intake has improved as much as we do the worker creation, so I would just focus on that for the introduction.

Author: I'd say the task intake is critical! They workers can start taking up tasks as soon as the worker is created, but really the former is the primary value add so I do want to emphasize that.

Previously, the number of Celery Worker replicas that could be immediately created following a deploy was restricted to 25% of the total number of desired Celery Workers - until the original Workers shut down. This meant that it took longer for replacement Celery Workers to be created and start picking up Airflow tasks. Defined as `maxSurge`, that percentage is now set to 100%, allowing the maximum number of Celery Worker replicas to exist and push your data pipelines forward.

> For users, this change will result in a quicker, more efficient and more graceful deploy process.

Reviewer 2: ```suggestion
For users, this change will result in a quicker, more efficient, and more graceful deploy process.
```

> > **Note:** This improvemenet is only applicable to those running Airflow with the Celery Executor on Astronomer. For more information on Airflow Executors, refer to ["Airflow Executors Explained"](https://www.astronomer.io/guides/airflow-executors-explained) or [Configure an Airflow Deployment on Astronomer](https://www.astronomer.io/docs/cloud/stable/deploy/configure-deployment).

Reviewer 2: ```suggestion
> **Note:** This improvement applies only to those running Airflow with the Celery Executor on Astronomer. For more information on Airflow Executors, refer to ["Airflow Executors Explained"](https://www.astronomer.io/guides/airflow-executors-explained) or [Configure an Airflow Deployment on Astronomer](https://www.astronomer.io/docs/cloud/stable/deploy/configure-deployment).
```

#### Bug Fixes & Improvements

- BugFix: Reset password flow triggers incorrect email for users who use local username/password to authenticate

### v0.21.1

> 

Reviewer 2: Doesn't look like it will let me suggest changes to unedited sections of the doc, but there are two additional typos here. Ctrl +f "Workpace" and "Deploment" to find them.

Author: Yikes good catch - applied change to Deployment typo but actually couldn't find Workspace typo. Will DM you to make sure I caught it!
